packages/haul-core/src/webpack/plugins/ReactNativeEnvPlugin.ts

```diff
@@ -0,0 +1,34 @@
+import webpack from 'webpack';
+
+const asyncEval = `
+// Fetch and eval async chunks
+function asyncEval(url) {
+  return fetch(url).then(res => res.text()).then(src => eval(src));
+}
+`;
+
+/**
+ * Adds React Native specific tweaks to bootstrap logic.
+ */
+export default class ReactNativeEnvPlugin {
+  apply(compiler: webpack.Compiler) {
+    compiler.hooks.thisCompilation.tap('ReactNativeEnvPlugin', compilation => {
+      (compilation.mainTemplate as any).hooks.bootstrap.tap(
+        'ReactNativeEnvPlugin',
+        (source: string) => {
+          return `${asyncEval}\n${source}`;
+        }
+      );
+
+      (compilation.mainTemplate as any).requireEnsure.tap(
+        'ReactNativeEnvPlugin',
+        (source: string) => {
+          return source.replace(
+            /importScripts\((.+)\)/gm,
+            'return asyncEval(__webpack_require__.p + $1)'
```

Comment:
is `return` necessary here?

Reply:
yes, it is extremely necessary

Reply:
a comment on why would be helpful, or a test showing before & after, especially that you replace all occurrences of `importScripts` (`g` flag in regex). I assume it's always at the end of some function block, but would be cool to make it more obvious :)